src/mem/alloc.h

```diff
@@ -158,36 +237,68 @@ namespace snmalloc
       return get(address_cast(p));
     }
 
+    /**
+     * Fetch a pointer through the pagemap.  The return value and the
+     * argument point at the same location, if the template parameter offset
+     * is true, but may have different metadata associated with them, such
+     * as CHERI bounds.  If the pagemap is storing pointers, then the
+     * template parameter offset may be false, in which case, the return
+     * value is exactly the pagemap entry, which will point at the bottom of
+     * the pagemap granule and include the argument in its bounds.
+     */
+    template<bool offset = true>
+    void* getp(void* p)
```

Comment:
Add fast path annotation, so we are guaranteed this will be erased in NON-Cheri case.